src/parser/monk/shared/modules/covenants/FallenOrder.tsx

```diff
@@ -0,0 +1,144 @@
+import { formatNumber } from 'common/format';
+import SpellUsable from 'parser/shared/modules/SpellUsable';
+import COVENANTS from 'game/shadowlands/COVENANTS';
+
+class FallenOrder extends Analyzer {
+  static dependencies = {
+    spellUsable: SpellUsable,
+    abilities: Abilities,
+  };
+
+  windDamage: number = 0;
+  brewDamage: number = 0;
+  mistDamage: number = 0;
+
+  mistHealing: number = 0;
+  mistOverhealing: number = 0;
+  cloneMap: Map<number, number> = new Map<number, number>();
+
+
+  constructor(options: any) {
+    super(options);
+
+    // In my testing this line always returns false but its what putput uses so I have faith
+    // it will work in the future. (might just be a bad log too)
+    this.active = this.selectedCombatant.hasCovenant(COVENANTS.VENTHYR.id);
+
+    if (!this.active) {
+      return;
+    }
+
+    options.abilities.add({
+      spell: SPELLS.FALLEN_ORDER_CAST,
+      category: Abilities.SPELL_CATEGORIES.COOLDOWNS,
+      cooldown: 180,
+      gcd: {
+        base: 1500,
+      },
+      castEfficiency: {
+        suggestion: true,
+        recommendedEfficiency: 0.8,
+      },
+    });
+
+    //summon events (need to track this to get melees)
+    this.addEventListener(Events.summon.by(SELECTED_PLAYER).spell([SPELLS.FALLEN_ORDER_OX_CLONE, SPELLS.FALLEN_ORDER_TIGER_CLONE, SPELLS.FALLEN_ORDER_CRANE_CLONE]), this.trackSummons);
+
+    //mistweaver spells
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER_PET).spell([SPELLS.FALLEN_ORDER_ENVELOPING_MIST, SPELLS.FALLEN_ORDER_SOOTHING_MIST]), this.mistHealingTracker);
+    //brewmaster spells
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER_PET).spell([SPELLS.FALLEN_ORDER_KEG_SMASH, SPELLS.FALLEN_ORDER_BREATH_OF_FIRE,SPELLS.BREATH_OF_FIRE_DEBUFF]), this.brewDamageTracker);
+    //windwalker spells
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER_PET).spell([SPELLS.FALLEN_ORDER_SPINNING_CRANE_KICK, SPELLS.FISTS_OF_FURY_DAMAGE]), this.windDamageTracker);
+    //shared
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER_PET).spell(SPELLS.MELEE), this.handleMelee);
+  }
+
+  trackSummons(event: SummonEvent){
+    this.cloneMap.set(event.targetID, event.ability.guid);
+  }
+
+  mistHealingTracker(event: HealEvent){
+    this.mistHealing += (event.amount || 0) + (event.absorbed || 0);
+    this.mistOverhealing += event.overheal || 0;
+  }
+
+  brewDamageTracker(event:DamageEvent){
+    this.brewDamage += event.amount || 0;
+  }
+
+  windDamageTracker(event:DamageEvent){
+    this.windDamage += event.amount || 0;
+  }
+
+  mistDamageTracker(event:DamageEvent){
+    this.mistDamage += event.amount || 0;
+  }
+
+  handleMelee(event: DamageEvent) {
+    //if we don't know who its from then we can't add it
+    if(!event.sourceID){
+      return;
+    }
+
+    const id: number = event.sourceID;
+    //if we are not tracking this guy we don't know what it is
+    if(!this.cloneMap.has(id)){
```

Comment:
You can remove this first check, you're doing it twice (checking for undefined after `.get`)

Reply:
Yeah. You're right